### PR TITLE
[FX-5726] Add base tailwind preset package

### PR DESCRIPTION
**Pull request by sashuk**
[FX-5726]

### Description

This pull request introduces the `@toptal/base-tailwind` package that provide base Tailwind configuration for any project. Configurations that are specific to projects are kept in the corresponding packages (for example, `@toptal/picasso-tailwind`).

### How to test

- [Temploy](https://picasso.toptal.net/fx-5726-create-base-preset-package)
- The https://github.com/toptal/staff-portal/pull/13505 applies both presets (please launch Staff Portal locally if needed). As you can see, the pull request temporary moves `fontWeight` to `@toptal/base-tailwind`, so we can see new font weight `700` to be applied from `@toptal/base-tailwind`.

Screenshot 2024-08-09 at 17 47 06

Some other styles are broken, as `fontWeight` is missing in `@toptal/picasso-tailwind` that some components rely on. 

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [N/A] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [N/A] Annotate all `props` in component with documentation
- [N/A] Create `examples` for component
- [N/A] Ensure that deployed demo has expected results and good examples
- [N/A] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [N/A] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-5762]: https://toptal-core.atlassian.net/browse/FX-5762?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[FX-5726]: https://toptal-core.atlassian.net/browse/FX-5726?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[B2-732]: https://toptal-core.atlassian.net/browse/B2-732?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by sashuk** [comment]
@toptal-anvil ping reviewers

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

File: tailwind.config.js [hunk 1]
```diff
@@ -1,10 +1,9 @@
 /** @type {import('tailwindcss').Config} */
 module.exports = {
   content: ['packages/**/*.{ts,tsx}'],
-  // Ignore missing "@toptal/picasso-tailwind" import as the package
-  // is definetely present in the project
+  // Ignore missing imports as packages are definitely present in the project
   // eslint-disable-next-line import/no-extraneous-dependencies
-  presets: [require('@toptal/picasso-tailwind')],
+  presets: [require('@toptal/base-tailwind'), require('@toptal/picasso-tailwind')],
```
Comment (sashuk) [inline tailwind.config.js:6]: - [ ] ⚠️ Revert this changes as well, as all Picasso packages needs to depend on both packages and have them as peer dependencies

**Comment by sashuk** [comment]
@toptal-anvil ping reviewers

File: .changeset/seven-flowers-retire.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+'@toptal/base-tailwind': patch
+'@toptal/picasso-tailwind': patch
```
Comment (denieler) [inline .changeset/seven-flowers-retire.md:3]: Maybe we can make this a major one so all users can update their usage and start using the base-tailwind preset. Because as soon as some styles are moved to this base preset from Picasso - it will be immediately a breaking change and it might be hard to spot potentially 🤔

File: packages/base-tailwind/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+Base Tailwind definitions. This preset has to be reused by any product-specific Tailwind configuration (Picasso, Caliber, etc.).
+
+
+```javascript
+/** @type {import('tailwindcss').Config} */
+module.exports = {
+  presets: [
+    require('@toptal/base-tailwind'),
+    require('@toptal/picasso-tailwind'),
+    require('@toptal/picasso-tailwind'),
```
Comment (denieler) [inline packages/base-tailwind/README.md:12]:
```suggestion
    require('@toptal/base-tailwind'),
    require('@toptal/picasso-tailwind'),
```

File: packages/picasso-tailwind/README.md [hunk 4]
```diff
@@ -1,3 +1,13 @@
 # `picasso-tailwind`
 
 Picasso's Tailwind definitions. Design tokens from BASE and support for Picasso usage.
+
+Please additionally use the `@toptal/base-tailwind` preset as a base for your Tailwind configuration.
```
Comment (denieler) [inline packages/picasso-tailwind/README.md:5]: I think this should be a default usage of this preset, so we can describe it just as a default example

File: packages/picasso-tailwind/README.md [hunk 5]
```diff
@@ -1,3 +1,13 @@
 # `picasso-tailwind`
 
 Picasso's Tailwind definitions. Design tokens from BASE and support for Picasso usage.
```
Comment (denieler) [inline packages/picasso-tailwind/README.md:3]:
```suggestion
Picasso-specific Tailwind definitions. Picasso-specific design tokens from BASE and support for Picasso usage.
```

**Review by denieler — COMMENTED** [review]
I think we should also change https://picasso.toptal.net/?path=/story/picasso-introduction--introduction to use base-tailwind from the start

File: packages/picasso-tailwind/README.md (hunk 5 above)
Comment (sashuk) [reply packages/picasso-tailwind/README.md]: @denieler Updated, thanks!

File: packages/picasso-tailwind/README.md (hunk 4 above)
Comment (sashuk) [reply packages/picasso-tailwind/README.md]: @denieler Updated, thanks!

File: packages/base-tailwind/package.json [hunk 6]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "@toptal/base-tailwind",
+  "version": "0.0.1",
+  "description": "Base Tailwind configuration, have to be reused by any project-specific Tailwind configuration",
```
[the remark this answers is not in the document]
Comment (sashuk) [reply packages/base-tailwind/package.json]: @denieler Added, thanks!

File: packages/base-tailwind/README.md (hunk 3 above)
Comment (sashuk) [reply packages/base-tailwind/README.md]: @denieler Fixed, thank you!

File: packages/base-tailwind/README.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
+# `base-tailwind`
+
+Base Tailwind configuration that contains global design token. This preset have to be reused by any project-specific Tailwind configuration (Picasso, Caliber, etc.).
```
Comment (mkrl) [inline packages/base-tailwind/README.md:3]:
```suggestion
BASE Tailwind configuration preset that contains global design tokens. This preset has to be reused by any project-specific Tailwind configuration (Picasso, Caliber, etc.).
```

**Review by denieler — APPROVED** [review]
Haven't tested, but the change looks good to me 👍 We need only to figure out how to deal with breaking-non-breaking changes before moving forward 🙇‍♂️

**Review by mkrl — APPROVED** [review]

File: .changeset/seven-flowers-retire.md (hunk 2 above)
Comment (sashuk) [reply .changeset/seven-flowers-retire.md:3]: @denieler Do you mean that `@toptal/picasso-tailwind` gets major update, but in fact the only change is that consumers have to manually add `@toptal/base-tailwind` preset to their Tailwind configurations (that use only `@toptal/picasso-tailwind` preset now)?

Comment (denieler) [reply .changeset/seven-flowers-retire.md:3]: @sashuk yes, but imagine we move 1 token from picasso-tailwind to base-tailwind, then all users, if they won't start using base-tailwind, would have a broken app.

File: packages/base-tailwind/README.md (hunk 7 above)
Comment (sashuk) [reply packages/base-tailwind/README.md]: @mkrl Addressed, thank you! I removed "configuration" from READMEs, as it is much more clear when the same definition ("preset") is used everywhere.

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

File: .changeset/seven-flowers-retire.md (hunk 2 above)
Comment (sashuk) [reply .changeset/seven-flowers-retire.md]: @denieler Agree, thanks! 

The `@toptal/picasso-tailwind` will have a major release with a note to install and reuse `@toptal/base-tailwind` preset in all consumer projects. However, the `@toptal/base-tailwind` is not a peer and any other dependency of `@toptal/picasso-tailwind`.

**Use case**

New token is introduced in `@toptal/base-tailwind`. One of the Picasso components `@toptal/picasso-input` now needs to depend on this token to implement some feature.

Solution: 

1. new minor version of `@toptal/base-tailwind` is released (with new token)
2. `@toptal/picasso-input` already has `@toptal/base-tailwind` as a peer dependency, but now it needs to bump the version of `@toptal/base-tailwind` peer dependency to match the version with introduced token
3. new major version of `@toptal/picasso-input` is released (due to change in peer dependencies)

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

**Comment by sashuk** [comment]
Staff Portal works as expected in https://github.com/toptal/staff-portal/pull/13505✅

**Comment by sashuk** [comment]
@toptal-anvil ping reviewers

**Review by TomasSlama — APPROVED** [review]
